### ISSUE TITLE
IFTTT not connecting to Glitch

### ISSUE THREAD

**Issue by dazazbab** [OmerTu/GoogleHomeKodi#305]
I've been using GoogleHomeKodi for a few months. No Issues until a couple days ago. Now IFTTT keep giving the error "Applet Skipped. Unable to make web request. Your server returned a 403" for any command. I didn't change anything in the IFTTT setup.  Google assistant responded but no actions happened in Kodi.

![Applet Activity](https://user-images.githubusercontent.com/66974089/84705564-97970c00-af29-11ea-98ba-b37aea3b54e4.png)

Glitch log doesn't even register that IFTTT trying to connect. I tried running some command from Glitch itself and it's running fine.

![Glitch_Test](https://user-images.githubusercontent.com/66974089/84705819-fe1c2a00-af29-11ea-880b-d4bedef8e6bc.png)

Any ideas what could cause this? It seems like a IFTTT/Webhook issue but I was wondering if anybody have seen this before.

**Comment by nousername8219** [comment]
Came to say the same thing, happening here as well. 

Works fine through Glitch. Thanks for the amazing application, I use it daily!

**Comment by RonnieBoy82** [comment]
Exactly the same issue here for a few days now. Test command works but nothing else

**Comment by corveck79** [comment]
Loving the app, but having the same issue. It started about 2 days ago.

**Comment by RonnieBoy82** [comment]
Get this in log when try to update from GitHub on glitch

“Could not find node ^13.9.0, using 10
1:06 AM
node v10.15.3, with pnpm”

If that helps any

**Comment by keydon** [comment]
Looks like glitch banned multiple "pingers" like IFTTT.
No idea if and when glitch will allow IFTTT again. (see https://support.glitch.com/t/ifttt-keeps-getiingt-403/)

**Comment by thewilloftheshadow** [comment]
Its a known problem, because of a Glitch outage. More information can be found here:
https://support.glitch.com/t/ping-service-block-june-13-7-56-a-m-to-present/26443

**Comment by keydon** [comment]
I can offer to share my personal instance hosted on google cloud run with you guys. If you guys are willing to trust me with your kodi-credentials of course (Hostname, Port, User and Password).
If interested just send me your details to `<my nickname here at github>`@gmail.com

**Comment by RiversideRocks** [comment]
This is not really intentional from Glitch, be more of an unintended side-effect of the ping service block. I'm really sorry this had to happen.

**Comment by terryo52404** [comment]
Would paying for the Glitch premium fix this?  I can't really tell if the response linked from the message above, about keeping apps awake would fix it. We only use it for Kodi but it is sorely missed now that it isn't working.

**Comment by nousername8219** [comment]
> This is not really intentional from Glitch, be more of an unintended side-effect of the ping service block. I'm really sorry this had to happen.
> 
> I suppose this would be the other option:
> https://support.glitch.com/t/ifttt-keeps-getiingt-403/26871/6

Thought I had lost my mind when I couldn't ping my server.

**Comment by RonnieBoy82** [comment]
Is there an alternative to glitch, with GitHub import?

**Comment by RonnieBoy82** [comment]
See on glitch support the ping block is permenent now

**Comment by corveck79** [comment]
> See on glitch support the ping block is permenent now. Am only able to use the glitch option for now and don't know about coding. Is there an alternative that is as easy to set up as glitch?

I would like too know as well.

**Comment by terryo52404** [comment]
I went ahead and paid for Glitch on a monthly basis and Kodi is working fine again and faster than it ever was before. But I am also interested in an easy to install alternative. I looked at the Google Cloud option and was thoroughly confused. If someone could put together a tutorial, that would be helpful.

**Comment by dazazbab** [comment]
I gave up on Glitch and went the Docker option. Got it set up and it was a lot more responsive than using Glitch. I did run to some issue since I wanted to use my edited version of the app (I previously edited en.json, server.js, and helpers.js on Glitch). I ended up copying those files into the Docker container in my RPi and everything is working perfectly.

**Comment by madmax2** [comment]
@dazazbab
which docker method did you use when editing the config?
Can you help me?
I ran a docker container with the edited settings but when I tried the commands with IFTTT
it shows socket hang up error message.

**Comment by dazazbab** [comment]
So I have edited en.json, server.js, and helpers.js when I was using glitch. When glitch messed up I switch to docker and tried to copy my edited files to the docker image and then run it. I never managed to get the it running. 

I ended up copying my edited files into the docker container after it has run. Then I restart the container and it worked.

For server.js and helpers.js I used the command 
docker cp [filepath] [container name]:/home/node/app/

For en.json I used the command
docker cp [filepath] [container name]:/home/node/app/broker/

To restart the container I used the command
docker restart [container name]

I figured out target path by going into the running container using the command
docker container exec -it [container name] sh

A lot of trial and error before finally got it to work.. 

Hope it works for you too..
